### ISSUE TITLE
gpg: Can't check signature: public key not found

### ISSUE THREAD

**Issue by ttresse** [pegasystems/docker-pega7-tomcat-ready#9]
Hi,
I'm trying to create few Docker images using this source code. The goal is to have on my local an image per version of tomcat. But before being able to do this, I can't build this image.

I clone this repository and run in the current repo: 
`docker build -t pegaready .`

The instructions in the Dockerfile runs until stop at the step 29/48 : 
`RUN curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL} -o catalina-jmx-remote.jar && curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL}.asc -o catalina-jmx-remote.jar.asc && gpg --verify catalina-jmx-remote.jar.asc && mv catalina-jmx-remote.jar /usr/local/tomcat/lib/catalina-jmx-remote.jar && rm catalina-jmx-remote.jar.asc
`
Here is the error: 

```
Step 29/48 : RUN curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL} -o catalina-jmx-remote.jar && curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL}.asc -o catalina-jmx-remote.jar.asc && gpg --verify catalina-jmx-remote.jar.asc && mv catalina-jmx-remote.jar /usr/local/tomcat/lib/catalina-jmx-remote.jar && rm catalina-jmx-remote.jar.asc
 ---> Running in 9c26dac3c27e
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13082  100 13082    0     0  17446      0 --:--:-- --:--:-- --:--:-- 17442
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   836  100   836    0     0   2683      0 --:--:-- --:--:-- --:--:--  2688
gpg: directory `/root/.gnupg' created
gpg: new configuration file `/root/.gnupg/gpg.conf' created
gpg: WARNING: options in `/root/.gnupg/gpg.conf' are not yet active during this run
gpg: keyring `/root/.gnupg/pubring.gpg' created
gpg: assuming signed data in `catalina-jmx-remote.jar'
gpg: Signature made Fri 29 Sep 2017 12:26:00 PM UTC using RSA key ID D63011C7
gpg: Can't check signature: public key not found
The command '/bin/sh -c curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL} -o catalina-jmx-remote.jar && curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL}.asc -o catalina-jmx-remote.jar.asc && gpg --verify catalina-jmx-remote.jar.asc && mv catalina-jmx-remote.jar /usr/local/tomcat/lib/catalina-jmx-remote.jar && rm catalina-jmx-remote.jar.asc' returned a non-zero code: 2
```

It seems to be a problem with the public key. Somebody can try to build this image to know if it's a general problem ?

Thank you in advance.

**Comment by viper0131** [comment]
Same issue

**Comment by cdancy** [comment]
@petehayes @dcasavant any thoughts on the above?

**Comment by viper0131** [comment]
I tried an older commit, but same results.

entire build log:
```
build "pega7-tomcat8-jre8-4" started
Cloning "https://github.com/pegasystems/docker-pega7-tomcat-ready.git" ...
	Commit:	f481f85b6bc79eaa3b52af66787f604031ad7457 (Improved wording via feedback)
	Author:	Peter Hayes <petehayes@gmail.com>
	Date:	Fri Jan 13 12:44:20 2017 -0500
Step 1 : FROM tomcat@sha256:90b2bee496f433123469c91586b19b6e4b1b0c417356ba0240bdcbea1b474a46
 ---> a92c139758db
Step 2 : RUN mkdir -p /opt/pega
 ---> Using cache
 ---> 98031f5c06cb
Step 3 : ENV DB_USERNAME pega
 ---> Using cache
 ---> dcd4a42a4381
Step 4 : ENV DB_PASSWORD pegasys
 ---> Using cache
 ---> b09480d5c64a
Step 5 : ENV DB_HOST postgresql
 ---> Using cache
 ---> 24aa3a31ad12
Step 6 : ENV DB_PORT 5432
 ---> Using cache
 ---> cb4571699364
Step 7 : ENV DB_NAME pega
 ---> Using cache
 ---> fe535bc8cf02
Step 8 : ENV JDBC_CLASS org.postgresql.Driver
 ---> Using cache
 ---> 8f35f2dc772d
Step 9 : ENV JDBC_DB_TYPE postgresql
 ---> Using cache
 ---> 71511dc426bd
Step 10 : ENV JDBC_URL_SUFFIX ''
 ---> Running in 8b7ccb5172b0
 ---> 9f59177d00d0
Removing intermediate container 8b7ccb5172b0
Step 11 : ENV JDBC_MIN_ACTIVE 50
 ---> Running in a6fabcac1863
 ---> 9858df377596
Removing intermediate container a6fabcac1863
Step 12 : ENV JDBC_MAX_ACTIVE 250
 ---> Running in 3832cdacb8f0
 ---> 3545f521397c
Removing intermediate container 3832cdacb8f0
Step 13 : ENV JDBC_MIN_IDLE 10
 ---> Running in 2ac25c57798b
 ---> ca47b29937eb
Removing intermediate container 2ac25c57798b
Step 14 : ENV JDBC_MAX_IDLE 50
 ---> Running in 951261de056e
 ---> d704edd7ed9c
Removing intermediate container 951261de056e
Step 15 : ENV JDBC_MAX_WAIT 30000
 ---> Running in bf8996f13c18
 ---> 8c697fea748e
Removing intermediate container bf8996f13c18
Step 16 : ENV JDBC_INITIAL_SIZE 50
 ---> Running in b836f64a6e41
 ---> d0229169ff14
Removing intermediate container b836f64a6e41
Step 17 : ENV RULES_SCHEMA pegarules
 ---> Running in deac63609578
 ---> cf0294accc2c
Removing intermediate container deac63609578
Step 18 : ENV DATA_SCHEMA pegadata
 ---> Running in ddf88a268428
 ---> 5c5400fd301b
Removing intermediate container ddf88a268428
Step 19 : ENV MAX_THREADS 300
 ---> Running in 2b3f005ee61d
 ---> bb33d68cf94f
Removing intermediate container 2b3f005ee61d
Step 20 : ENV JAVA_OPTS -Xms2048m -Xmx4096m -XX:PermSize=64m -XX:MaxPermSize=384m
 ---> Running in fba5c0fe6917
 ---> 7f12681f6773
Removing intermediate container fba5c0fe6917
Step 21 : ENV INDEX_DIRECTORY NONE
 ---> Running in a2d7791998b6
 ---> fd0875f7bdac
Removing intermediate container a2d7791998b6
Step 22 : ENV HEAP_DUMP_PATH /heapdumps
 ---> Running in 09a3d583c441
 ---> e976c8a86211
Removing intermediate container 09a3d583c441
Step 23 : ENV NODE_ID NONE
 ---> Running in 6f8548522be3
 ---> 111c6ec49f00
Removing intermediate container 6f8548522be3
Step 24 : ENV JMX_PORT 9001
 ---> Running in 36594fb73a88
 ---> 927f229ab6a4
Removing intermediate container 36594fb73a88
Step 25 : ENV JMX_SERVER_HOSTNAME 127.0.0.1
 ---> Running in 96f02d411ba9
 ---> 2a60f60945e0
Removing intermediate container 96f02d411ba9
Step 26 : ENV TOMCAT_JMX_JAR_TGZ_URL https://archive.apache.org/dist/tomcat/tomcat-$TOMCAT_MAJOR/v$TOMCAT_VERSION/bin/extras/catalina-jmx-remote.jar
 ---> Running in 5e8e40f5500b
 ---> 96da58eff53b
Removing intermediate container 5e8e40f5500b
Step 27 : RUN curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL} -o catalina-jmx-remote.jar && curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL}.asc -o catalina-jmx-remote.jar.asc && gpg --verify catalina-jmx-remote.jar.asc && mv catalina-jmx-remote.jar /usr/local/tomcat/lib/catalina-jmx-remote.jar && rm catalina-jmx-remote.jar.asc
 ---> Running in 9c13f71a67ad
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13618  100 13618    0     0  81220      0 --:--:-- --:--:-- --:--:-- 82533
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   188  100   188    0     0   1758      0 --:--:-- --:--:-- --:--:--  1773
gpg: directory '/root/.gnupg' created
gpg: keybox '/root/.gnupg/pubring.kbx' created
gpg: assuming signed data in 'catalina-jmx-remote.jar'
gpg: Signature made Thu Jan 18 20:13:04 2018 UTC
gpg:                using DSA key 6FB21E8933C60243
gpg: Can't check signature: No public key
error: build error: The command '/bin/sh -c curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL} -o catalina-jmx-remote.jar && curl -kSL ${TOMCAT_JMX_JAR_TGZ_URL}.asc -o catalina-jmx-remote.jar.asc && gpg --verify catalina-jmx-remote.jar.asc && mv catalina-jmx-remote.jar /usr/local/tomcat/lib/catalina-jmx-remote.jar && rm catalina-jmx-remote.jar.asc' returned a non-zero code: 2
error: the build 00-build/pega7-tomcat8-jre8-4 status is "Failed"```

**Comment by viper0131** [comment]
/root/.gnupg dir is no longer in the tomcat base image?

Recent Image
```
$ docker run -ti tomcat:8-jre8 ls -al /root
total 20
drwx------ 1 root root 4096 Oct 10 08:40 .
drwxr-xr-x 1 root root 4096 Jan 28 22:02 ..
-rw-r--r-- 1 root root  570 Jan 31  2010 .bashrc
drwx------ 1 root root 4096 Oct 10 08:40 .gnupg
-rw-r--r-- 1 root root  148 Aug 17  2015 .profile
```

Older image, still on my host:

```
$ docker run -ti tomcat:7-jre7 ls -al /root
total 16
drwx------ 2 root root 4096 Dec 10 00:00 .
drwxr-xr-x 1 root root 4096 Jan 28 22:02 ..
-rw-r--r-- 1 root root  570 Jan 31  2010 .bashrc
-rw-r--r-- 1 root root  140 Nov 19  2007 .profile
```

**Comment by viper0131** [comment]
[latest commit on tomcat dockerfile on github](https://github.com/docker-library/tomcat/commit/5d317a0a21e6c927e415dfc8dd452619ea4b4643#diff-cfb992ab7a10b37f678cc3e6f012f127)

the changed a few things around gpg... probably the root cause of this issue.

**Comment by viper0131** [comment]
@cdancy Can you please check this out. Currently this breaks the build of the image.

**Comment by dcasavant** [comment]
Merged #11

**Comment by viper0131** [comment]
Thanks, confirmed that I can build the image again!
